File: AssetManager.Api/Models/ExportedAssetFile.cs
using AssetManager.Api.Models.Enums;
using AssetManager.Api.Models;

public class ExportedAssetFile
{
    public Guid Id { get; set; }

    public Guid Asset3DId { get; set; }
    public Asset3D Asset3D { get; set; } = null!;

    public string Format { get; set; } = string.Empty;
    public string FileName { get; set; } = string.Empty;
    public string FilePath { get; set; } = string.Empty;

    public long FileSize { get; set; }

    public ExportStatus Status { get; set; } = ExportStatus.Pending;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: asset-manager-api-node/src/domain/entities/ExportedAssetFile.ts
import { ExportStatus } from "../enums/ExportStatus";

export class ExportedAssetFile {
  constructor(
    public readonly id: string,
    public readonly assetId: string,
    public format: string,
    public fileName: string,
    public filePath: string,
    public fileSize: number,
    public status: ExportStatus,
    public readonly createdAt: Date
  ) {}
}
